### PR TITLE
Test for cross shard transactions

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  override fun buildSQLExceptionConversionDelegate(): SQLExceptionConversionDelegate {
+    val superDelegate = super.buildSQLExceptionConversionDelegate()
+    return SQLExceptionConversionDelegate { sqlException, message, sql ->
+      val exceptionMessage = sqlException.message
+      if (exceptionMessage != null && exceptionMessage.contains("multi-db transaction attempted")) {
+        throw CrossShardTransactionException(message, sqlException)
+      }
+
+      when (sqlException.errorCode) {
+        1205 -> {
```
Comment (eeSeeGee) [inline misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt]: Can these be constified somewhere?

File: misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          return@SQLExceptionConversionDelegate PessimisticLockException(message, sqlException, sql)
+        }
+        1207, 1206 -> {
+          return@SQLExceptionConversionDelegate LockAcquisitionException(message, sqlException, sql)
+        }
+      }
+
+      val sqlState = JdbcExceptionHelper.extractSqlState(sqlException)
+
+      if ("41000" == sqlState) {
```
Comment (eeSeeGee) [inline misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt]: Ditto here

File: misk-hibernate/src/test/kotlin/misk/hibernate/ScaleSafetyTest.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+class ScaleSafetyTest {
+  @MiskTestModule
+  val module = MoviesTestModule()
+
+  @Inject @Movies lateinit var transacter: Transacter
+  @Inject lateinit var queryFactory: Query.Factory
+  @Inject lateinit var tracer: MockTracer
+
+  @Test
+  fun crossShardTransactionsAreDisabled() {
```
Comment (eeSeeGee) [inline misk-hibernate/src/test/kotlin/misk/hibernate/ScaleSafetyTest.kt]: Maybe test for updates too?

File: misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt (hunk 1 above)
Comment (tirsen) [reply misk-hibernate/src/main/kotlin/misk/hibernate/VitessDialect.kt]: ew wait this is all in the baseclass

**Pull request merged**